jstest/testrunner/devices/stm32f4dis.py

```diff
@@ -30,13 +32,17 @@ def __init__(self, env):
         RemoteDevice.__init__(self, env)
 
         data = {
-            'dev-id': env['info']['device_id'],
-            'baud': env['info']['baud'],
             'timeout': env['info']['timeout'],
             'prompt': 'nsh> '
         }
 
-        self.channel = SerialConnection(data)
+        if self.ip:
+            data['ip'] = self.ip
+            self.channel = TelnetConnection(data)
+        else:
+            data['dev-id'] = env['info']['device_id']
+            data['baud'] = env['info']['baud']
+            self.channel = SerialConnection(data)
```

Comment:
In the `__init__.py` file in the communication folder, you can create a `create_connection` method that could return the appropriate connection object (based on that IP address is provided, or not). Similar solution can be found in the [devices/__init__.py](https://github.com/Samsung/js-remote-test/blob/master/jstest/testrunner/devices/__init__.py) file.